lib/internal/modules/package_json_reader.js

```diff
@@ -104,29 +117,56 @@ function readPackage(requestPath) {
 /**
  * Get the nearest parent package.json file from a given path.
  * Return the package.json data and the path to the package.json file, or undefined.
- * @param {string} checkPath The path to start searching from.
- * @returns {undefined | {data: import('typings/internalBinding/modules').PackageConfig, path: string}}
+ * @param {URL['href'] | URL['pathname']} startPath The path to start searching from.
+ * @param {boolean} everything Whether to include the full contents of the package.json.
+ * @returns {undefined | {
+ *   data: everything extends true ? FullPackageConfig : RecognizedPackageConfig,
+ *   path: URL['pathname'],
+ * }}
  */
-function getNearestParentPackageJSON(checkPath) {
-  const result = modulesBinding.getNearestParentPackageJSON(checkPath);
+function getNearestParentPackageJSON(startPath, everything = false) {
+  if (typeof startPath !== 'string') {
+    throw new ERR_INVALID_ARG_TYPE('startPath', 'string', startPath);
+  }
+  if (
+    everything !== undefined &&
+    typeof everything !== 'boolean'
+  ) {
```

Comment:
I'm pretty sure there's no way for `everything` to be `undefined` because of the default value.
```suggestion
  if (typeof everything !== 'boolean') {
```